scvi/module/_jaxvae.py

```diff
@@ -146,7 +142,7 @@ def _get_inference_input(self, tensors: Dict[str, jnp.ndarray]):
         return input_dict
 
     def inference(self, x: jnp.ndarray, n_samples: int = 1) -> dict:
-        mean, var = self.encoder(x)
+        mean, var = self.encoder(x, is_training=self.is_training)
```

Comment:
I prefer if is_training is pushed through the inference and generative functions. Right now it would require new JaxModules to handle is_training state on its own. Ideally, the Flax module should not hold any state at all and remain entirely pure. This is why we have the JaxModuleWrapper to hold state and wrap the flax module.